src/decode.rs

```diff
@@ -3447,28 +3516,29 @@ unsafe fn decode_sb(
     let frame_hdr = &***f.frame_hdr.as_ref().unwrap();
 
     let bp;
-    let mut ctx = 0;
     let mut bx8 = 0;
     let mut by8 = 0;
-    let pc = if t.frame_thread.pass == 2 {
+    let ctx = if t.frame_thread.pass == 2 {
         None
     } else {
         if false && bl == BlockLevel::Bl64x64 {
+            let ts_c = ts.context.try_lock().unwrap();
             println!(
                 "poc={},y={},x={},bl={:?},r={}",
-                frame_hdr.frame_offset, t.b.y, t.b.x, bl, ts.msac.rng,
+                frame_hdr.frame_offset, t.b.y, t.b.x, bl, ts_c.msac.rng,
             );
         }
         bx8 = (t.b.x & 31) >> 1;
         by8 = (t.b.y & 31) >> 1;
-        ctx = get_partition_ctx(&f.a[t.a], &t.l, bl, by8, bx8);
-        Some(&mut ts.cdf.m.partition[bl as usize][ctx as usize])
+        Some(get_partition_ctx(&f.a[t.a], &t.l, bl, by8, bx8) as usize)
```

Comment:
```suggestion
        Some(get_partition_ctx(&f.a[t.a], &t.l, bl, by8, bx8))
```

I think it's a bit better to store it as an `Option<u8>` than an `Option<usize>` I think, and do the `as usize` casts at use site, since it's smaller.  It could be optimized, but this is a big function.